gems/aws-sdk-core/lib/aws-sdk-core/credentials_provider_chain.rb

```diff
@@ -2,91 +2,104 @@
 
 module AWS::SDK::Core
   # @api private
-  module CredentialsProviderChain
-    def self.call(config)
+  class CredentialsProviderChain < Hearth::IdentityProvider
+    def initialize(config)
+      @config = config
+      super(nil)
+    end
+
+    def identity(_properties = {})
+      return @identity if @identity
```

Comment:
Should this be a refreshing credential provider? Right now I think it will just cache the first identity and then never refresh?